baseplate/experiments/providers/__init__.py

```diff
@@ -55,28 +70,31 @@ def parse_experiment(config):
     :return: A subclass of :py:class:`Experiment` for the given experiment
         type.
     """
-    experiment_type = config["type"].lower()
-    experiment_id = config["id"]
-    assert isinstance(experiment_id, int)
-    name = config["name"]
+    experiment_type = config.get("type")
+    if experiment_type:
+        experiment_type = experiment_type.lower()
+    experiment_id = config.get("id")
+    if not isinstance(experiment_id, int):
+        raise KeyError("Integer id must be provided for experiment.")
```

Comment:
It seems weird to raise a `KeyError` if experiment_id happens to be a string or something like that, `KeyError` only really makes sense if `"id"` isn't in the config so `experiment_id` would be `None`.

Reply:
`TypeError` would be the more appropriate built-in exception